services/legacy-service.js

```diff
@@ -8,13 +8,11 @@ class LegacyService extends BaseService {
     throw Error(`registerLegacyRouteHandler() not implemented for ${this.name}`)
   }
 
-  static register(
-    { camp, handleRequest: cache, githubApiProvider },
-    serviceConfig
-  ) {
+  static register({ camp, handleRequest, githubApiProvider }, serviceConfig) {
+    const { cache: cacheHeaderConfig } = serviceConfig
     this.registerLegacyRouteHandler({
       camp,
-      cache,
+      cache: (...args) => handleRequest(cacheHeaderConfig, ...args),
```

Comment:
This is hard to read, what is happening on this line?

Reply:
At a high level, it's partially applying the `cacheHeaderConfig` parameter to the `handleRequest` function, returning a new function. The new function can be called with any number of parameters, and they will all be forwarded to `handleRequest`, though `cacheHeaderConfig` will be inserted at the beginning.

This is pretty ugly, though I've done it here because this is legacy code stapled on top of other legacy code providing a consistent interface to even more legacy code… so it's both temporary, and not really expected to be be pretty.

I just searched for a good explanation of partial application but almost everything I'm finding is filled with functional-programming jargon that makes the concept seem very complicated.

This article gives a pretty good explanation, and a simple example that motivates why you'd want it:

https://scotch.io/tutorials/javascript-functional-programming-explained-partial-application-and-currying

He's using Ramda:

```js
const buildHttpsUri = R.partial(buildUri, ['https'])
```

Though that requires even _more_ context to understand than the native ES6 equivalent:

```js
const buildHttpsUri = (...args) => buildUri('https', ...args)
```

Using [rest parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters#Description) and [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), it works regardless of the number of arguments. Though if you knew the number of arguments and their names, you could write it more explicitly:

```js
const buildHttpsUri = (foo, bar, baz) => buildUri('https', foo, bar, baz)
```